File: src/test/resources/FibFast.java
public class Solution {
    /**
     * @param n id of fibonacci term to be returned
     * @return  N'th term of Fibonacci sequence
     */
    public long fib(int n) {
        long a = 0;
        long b = 1;

        for (int i = 31 - Integer.numberOfLeadingZeros(n); i >= 0; i--) {
            long d = a * ((b<<1) - a);
            long e = (a*a) + (b*b);
            a = d;
            b = e;
            if (((n >>> i) & 1) != 0) {
                long c = a+b;
                a = b;
                b = c;
            }
        }

        return a;
    }
}

File: src/test/resources/FibFast.kt
class FibFast {
    /**
     * @param n id of fibonacci term to be returned
     * *
     * @return  N'th term of Fibonacci sequence
     */
    fun fib(n: Int): Long {
        var a: Long = 0
        var b: Long = 1

        for (i in 31 - Integer.numberOfLeadingZeros(n) downTo 0) {
            val d = a * ((b shl 1) - a)
            val e = a * a + b * b
            a = d
            b = e
            if (n.ushr(i) and 1 != 0) {
                val c = a + b
                a = b
                b = c
            }
        }

        return a
    }
}